From 9ce0de9ac9d4ed84dc89daffe6d1a247d388a4dc Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 27 Jan 2022 21:49:43 +0200
Subject: [PATCH] [SW-69188] gaudi3: initialize sku information

ROM needs sku information to perform certain die to die configuration.
This patch add support to read sku information from fuse and store
it in local structure.

Change-Id: Ib1194d254b44e2ef6595403539225326063e1c70
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 54b8432..8b7adbe 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -22,7 +22,8 @@
 	.efuse_read = 0,
 	.curr_img = 0,
 	.erom_boot_desc_flag = 0,
-	.die_id = 0
+	.die_id = 0,
+	.sku = 0
 };
 
 #define ARCBOOT_BRINGUP_ERROR		-1
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 7ce19ad..36a37ff 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -145,4 +145,7 @@
 		val = EFUSE_SYSCFG_DEFAULT;
 
 	arcboot_rom_global.config_fuse = val;
+
+	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SKU_MASK)
+		arcboot_rom_global.sku = 1;
 }
diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index 70bdf04..206349a 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -28,6 +28,8 @@
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI		BIT(8)
 #define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI		BIT(8)
 
+#define EFUSE_SYSCFG_SKU_MASK				BIT(26)
+
 /*
  * As per recommendation from Security Architect, if FUSE is not available,
  * card should not continue in the secure boot mode. Hence the default
diff --git a/include/arcboot.h b/include/arcboot.h
index 6f14e07..80d380e 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -66,6 +66,7 @@
 	uint8_t curr_img;
 	uint8_t erom_boot_desc_flag;
 	uint8_t die_id;
+	uint8_t sku; /* 0: Dual die 1: Single die */
 };
 
 enum arcboot_rom_bringup_proggress {
